FWCore/Framework/interface/ProducesCollector.h

```diff
@@ -86,5 +97,27 @@ namespace edm {
     propagate_const<ProductRegistryHelper*> helper_;
   };
 
+  template <Transition B>
+  class ProducesCollectorAdaptor {
+  public:
+    using Adapter = ProducesCollectorAdaptor<B>;
+
+    template <typename TYPE>
+    EDPutTokenT<TYPE> produces() {
+      return m_producer.template produces<TYPE, B>(m_label);
+    }
+
+  private:
+    //only ConsumesCollector is allowed to make an instance of this class
```

Comment:
Should this be 
```suggestion
    //only ProducesCollector is allowed to make an instance of this class
```
?